src/Sarif/Baseline/ResultMatching/SarifLogResultMatcher.cs

```diff
@@ -343,5 +347,24 @@ private List<ExtractedResult> ExtractResultsFromRuns(IEnumerable<Run> sarifRuns)
                 }
             }
         }
+
+        private static IList<T> MergeList<T>(IEnumerable<T> source1, IEnumerable<T> source2, IEqualityComparer<T> comparer)
```

Comment:
>MergeList [](http://example.com/codeflow?start=32&length=9)

I don't see unit tests for this generic helper.

This is extremely generic code, do you have it in the right place? Should it live elsewhere?

Reply:
refactor the function and unit tests added pls see the latest change.